packages/components/src/mobile/bottom-sheet/bottom-sheet-navigation/test/navigation-container.native.js

```diff
@@ -27,7 +27,13 @@ const TestScreen = ( { fullScreen, name, navigateTo } ) => {
 	);
 };
 
-jest.useFakeTimers();
+beforeAll( () => {
```

Comment:
Have you tried moving `jest.useFakeTimers( 'legacy' );` after import statements? Jest isolates everything per test file so you probably don't need to wrap it with lifecycle methods and revert setup. This is what I have in #33287. Anyway, it's fine as is in case you prefer it.

Reply:
> Have you tried moving `jest.useFakeTimers( 'legacy' );` after import statements? Jest isolates everything per test file so you probably don't need to wrap it with lifecycle methods and revert setup. 

By "after import statements" I interpret you to mean dropping the `(before|after)All` and `jest.useRealTimers()` calls, i.e. merely invoke `jest.useFakeTimers("legacy")` top-level in the file? I believe you are correct we could revert the additions of  `(before|after)All` and `jest.useRealTimers()`. Those additions were done during explorations of resolving the original errors and likely unnecessary. I will push a commit reverting those unnecessary changes.

Reply:
> i.e. merely invoke jest.useFakeTimers("legacy") top-level in the file?

Yes, this is exactly what I meant. It's my favorite feature of Jest that changes applied to the test framework in a single file don't bubble up to other files. The same applies to Jest mocks and other APIs.